src/any_agent/evaluation/evaluation_runner.py

```diff
@@ -0,0 +1,169 @@
+from textwrap import dedent
+
+from any_agent.config import AgentFramework
+from any_agent.evaluation.evaluation_case import EvaluationCase
+from any_agent.evaluation.evaluators import (
+    CheckpointEvaluator,
+    HypothesisEvaluator,
+    QuestionAnsweringSquadEvaluator,
+)
+from any_agent.evaluation.evaluators.schemas import EvaluationResult
+from any_agent.evaluation.results_saver import save_evaluation_results
+from any_agent.logging import logger
+from any_agent.tracing import TracingProcessor
+from any_agent.tracing.trace import AgentTrace
+
+
+class EvaluationRunner:
+    def __init__(self, output_path: str = "output") -> None:
+        self._evaluation_cases: list[EvaluationCase] = []
+        self._traces: list[AgentTrace] = []
+        self._trace_frameworks: list[AgentFramework] = []
+        self.checkpoint_evaluator: CheckpointEvaluator | None = None
+        self.hypothesis_evaluator: HypothesisEvaluator | None = None
+        self.qa_evaluator: QuestionAnsweringSquadEvaluator | None = None
+        self.output_path: str = output_path
+
+    def _setup_evaluators(self, evaluation_case: EvaluationCase) -> None:
+        self.checkpoint_evaluator = CheckpointEvaluator(model=evaluation_case.llm_judge)
+        self.hypothesis_evaluator = HypothesisEvaluator(model=evaluation_case.llm_judge)
+        self.qa_evaluator = QuestionAnsweringSquadEvaluator()
+
+    def add_evaluation_case(self, evaluation_case: EvaluationCase) -> None:
+        """Add test case file path to the evaluation runner."""
+        if evaluation_case not in self._evaluation_cases:
+            self._evaluation_cases.append(evaluation_case)
+        else:
+            logger.warning("Test case %s already added.", evaluation_case)
+
+    def add_trace(self, trace: AgentTrace, agent_framework: AgentFramework) -> None:
+        """Add telemetry file path to the evaluation runner."""
+        if trace not in self._traces:
+            self._traces.append(trace)
+            self._trace_frameworks.append(agent_framework)
+        else:
+            logger.warning("Telemetry %salready added.", trace)
+
+    def _run_trace_eval(
+        self,
+        evaluation_case: EvaluationCase,
+        trace: AgentTrace,
+        agent_framework: AgentFramework,
+    ) -> None:
+        processor = TracingProcessor.create(agent_framework)
```

Comment:
not sure if we already discuss this, but ideally here we would not need to import `TracingProcessor` and have the extract logic defined within `evaluation`